vector/src/Data/Vector/Unboxed/Base.hs

```diff
@@ -764,6 +768,139 @@ instance (Unbox a, Unbox b) => G.Vector Vector (Arg a b) where
   elemseq _ (Arg x y) z          = G.elemseq (undefined :: Vector a) x
                                  $ G.elemseq (undefined :: Vector b) y z
 
+-- -------
+-- Unboxing the boxed values
+-- -------
+
+-- | Newtype which allows to derive unbox instances for type @a@ which
+-- is normally a "boxed" type. The newtype stictly evaluates the wrapped values
+-- via thier requisite 'NFData' instance, ensuring that the unboxed vector
+-- contains only values reduced to normal form.
+-- For a strict newtype wrapper, see 'AsBoxedStrictly'.
+--
+-- 'AsBoxedStrictly' is intended to be unsed in conjunction with the newtype 'As'
+-- and the type class 'IsoUnbox'. Here's an example which uses the following
+-- explicit 'IsoUnbox' instance:
+--
+--
+-- >>> :set -XTypeFamilies -XStandaloneDeriving -XDerivingVia
+-- >>> :set -XMultiParamTypeClasses -XTypeOperators -XFlexibleInstances
+-- >>> import qualified Data.Vector.Unboxed         as VU
+-- >>> import qualified Data.Vector.Unboxed.Mutable as VUM
+-- >>> import qualified Data.Vector.Generic         as VG
+-- >>> import qualified Data.Vector.Generic.Mutable as VGM
+-- >>> :{
+-- data Foo a = Foo Int a
+--   deriving Show
+-- instance NFData a => VU.IsoUnbox (Foo a) (Int, AsBoxedStrictly a) where
+--   toURepr (Foo i a) = (i, makeBoxedStrictly a)
+--   fromURepr (i, a) = Foo i $ grabBoxedStrictly a
+--   {-# INLINE toURepr #-}
+--   {-# INLINE fromURepr #-}
+-- newtype instance VU.MVector s (Foo a) = MV_Foo (VU.MVector s (Int, AsBoxedStrictly a))
+-- newtype instance VU.Vector    (Foo a) = V_Foo  (VU.Vector    (Int, AsBoxedStrictly a))
+-- deriving via (Foo a `VU.As` (Int, AsBoxedStrictly a)) instance NFData a => M.MVector VUM.MVector (Foo a)
+-- deriving via (Foo a `VU.As` (Int, AsBoxedStrictly a)) instance NFData a => VG.Vector   VU.Vector   (Foo a)
+-- instance NFData a => VU.Unbox (Foo a)
+-- :}
+-- >>> VU.fromListN 3 [ Foo 4 "Haskell's", Foo 8 "strong", Foo 16 "types" ]
+-- [ Foo 4 "Haskell's", Foo 8 "strong", Foo 16 "types" ]
+--
+-- @since 0.13.2.0n
+newtype AsBoxedStrictly a = AsBoxedStrictly a
+
+-- | Smart constructor for creating values of the 'AsBoxedStrictly' newtype.
+-- The wrapped values are evaluated to normal form via thier 'NFData' instance.
+--
+-- @since 0.13.2.0
```

Comment:
No reason to add constructor/accessor. Newtype's constructor should be exported anyway in order for deriving via to work.

 Anyway we aren't ensuring strictness in wrapper newtype. We're doing it in type class methods. Newtype is going to be coerced away. It's only used to pick instance.

Reply:
I've updated the API accordingly.